File: write_data.m
function write_data(output,particlelist,params)
population=particlelist.Nprime;
fileID = fopen([output.path output.file],'w');
fprintf(fileID,'ITEM: TIMESTEP\n0\nITEM: NUMBER OF ATOMS\n%d\n',population);
fprintf(fileID,'ITEM: BOX BOUNDS pp pp pp\n%f %f\n%f %f\n%f %f\n',-params.boxlength/2,params.boxlength/2,-params.boxlength/2,params.boxlength/2,-params.boxlength/2,params.boxlength/2);
fprintf(fileID,'ITEM: ATOMS id type x y z a b c qw qx qy qz\n');
Alldata=[(1:population)' particlelist.xyz(1:population,:) particlelist.ac(1:population,[1 1 2]) particlelist.quat(1:population,:)];
fprintf(fileID,'%d 1 %f %f %f %f %f %f %f %f %f %f\n',Alldata');
fclose(fileID);
end

File: python_code/write_data.py
def write_data(output, particlelist, params):
    population = particlelist['Nprime']
    fileID = open(output['path'] + output['file'], 'w')
    fileID.write('ITEM: TIMESTEP\n0\n')
    fileID.write('ITEM: NUMBER OF ATOMS\n%d\n' % population)
    fileID.write('ITEM: BOX BOUNDS pp pp pp\n%f %f\n%f %f\n%f %f\n' % (-params['boxlength']/2, params['boxlength']/2, -params['boxlength']/2, params['boxlength']/2, -params['boxlength']/2, params['boxlength']/2))
    fileID.write('ITEM: ATOMS id type x y z a b c qw qx qy qz\n')
    Alldata = [(i+1, particlelist['xyz'][i][0], particlelist['xyz'][i][1], particlelist['xyz'][i][2], particlelist['ac'][i][0], particlelist['ac'][i][0], particlelist['ac'][i][1], particlelist['quat'][i][0], particlelist['quat'][i][1], particlelist['quat'][i][2], particlelist['quat'][i][3]) for i in range(population)]
    for data in Alldata:
        fileID.write('%d 1 %f %f %f %f %f %f %f %f %f %f\n' % data)
    
    fileID.close()
